src/functions/views/provider.php

```diff
@@ -17,3 +17,20 @@ function tribe_events_tickets_views_v2_is_enabled() {
 
 	return tribe_events_views_v2_is_enabled();
 }
+
+/**
+ * Normalizes the post ID by converting provisional ID's into regular Post ID's, if necessary.
+ * 
+ * @since TBD
+ *
+ * @param int $post_id The Post ID to normalize.
+ *
+ * @return int The normalized post ID.
+ */
+function tribe_events_tickets_normalize_post_id( $post_id ) {
+	if ( class_exists( '\TEC\Events\Custom_Tables\V1\Models\Occurrence', false ) ) {
+		return \TEC\Events\Custom_Tables\V1\Models\Occurrence::normalize_id( $post_id );
+	}
+
+	return $post_id;
+}
```

Comment:
We already have a filter for this here: https://github.com/the-events-calendar/event-tickets/blob/2198626e84d17071a1fe9d6e17f2e2f0b540c9e0/src/Tickets/Custom_Tables/V1/Provider.php#L51

So, the wrapper method can just contain the filter and won't need to check the class anymore. Also, we are moving away from `tribe` namespace so, the method can be named the same as the filter.


```suggestion
function tec_tickets_filter_event_id( $post_id ) {
	return apply_filters( 'tec_tickets_filter_event_id', $post_id );
}
```

Reply:
@rafsuntaskin I've updated the function name to use the `tec` prefix. Using the filter in this scenario doesn't normalize the `$post_id`. I think there might be an issue with the filter execution order.